### PR TITLE
Adds simple Python to wait for Redis broker to be ready

### LINKED ISSUE #780: [BUG] Paperless does not try to reconnect to Redis if it fails on startup
### Description

If Redis is not booted up yet or the container is restarting (especially in Kubernetes), paperless does not try to reconnect to it and therefore can't process any scheduled tasks like mail fetching.

### Expected behavior

paperless is trying to (re-)connect to Redis on startup until it can reach Redis or kills itself if it fails too often.

### Steps to reproduce

1. Start a paperless instance without Redis
2. You're done. Paperless seems to work (at least UI etc.) but task processing will not work.

### Webserver logs

```bash
➜  ~ k logs -n paperless-prod paperless-prod-0 --tail=50
Executing /usr/local/bin/supervisord -c /etc/supervisord.conf
2022-04-20 11:46:56,761 INFO Set uid to user 0 succeeded
2022-04-20 11:46:56,769 INFO supervisord started with pid 1
2022-04-20 11:46:57,772 INFO spawned: 'consumer' with pid 290
2022-04-20 11:46:57,775 INFO spawned: 'gunicorn' with pid 291
2022-04-20 11:46:57,779 INFO spawned: 'scheduler' with pid 292
[2022-04-20 13:46:58 +0200] [291] [INFO] Starting gunicorn 20.1.0
[2022-04-20 13:46:58 +0200] [291] [INFO] Listening at: http://0.0.0.0:8000 (291)
[2022-04-20 13:46:58 +0200] [291] [INFO] Using worker: paperless.workers.ConfigurableWorker
[2022-04-20 13:46:58 +0200] [291] [INFO] Server is ready. Spawning workers
13:46:59 [Q] INFO Q Cluster hotel-cat-kentucky-tennessee starting.
2022-04-20 11:46:59,590 INFO success: consumer entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2022-04-20 11:46:59,590 INFO success: gunicorn entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2022-04-20 11:46:59,590 INFO success: scheduler entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
13:46:59 [Q] ERROR Can not connect to Redis server.
Process Process-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/redis/connection.py", line 559, in connect
    sock = self._connect()
  File "/usr/local/lib/python3.9/site-packages/redis/connection.py", line 615, in _connect
    raise err
  File "/usr/local/lib/python3.9/site-packages/redis/connection.py", line 603, in _connect
    sock.connect(socket_address)
OSError: [Errno 113] No route to host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/django_q/cluster.py", line 168, in __init__
    self.start()
  File "/usr/local/lib/python3.9/site-packages/django_q/cluster.py", line 171, in start
    self.broker.ping()
  File "/usr/local/lib/python3.9/site-packages/django_q/brokers/redis_broker.py", line 39, in ping
    raise e
  File "/usr/local/lib/python3.9/site-packages/django_q/brokers/redis_broker.py", line 36, in ping
    return self.connection.ping()
  File "/usr/local/lib/python3.9/site-packages/redis/client.py", line 1378, in ping
    return self.execute_command('PING')
  File "/usr/local/lib/python3.9/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/usr/local/lib/python3.9/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/usr/local/lib/python3.9/site-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error 113 connecting to paperless-redis:6379. No route to host.
[2022-04-20 13:46:59,762] [INFO] [paperless.management.consumer] Using inotify to watch directory for changes: /usr/src/paperless/src/../consume
```

### Screenshots

_No response_

### Paperless-ngx version

1.6.0

### Host OS

Ubuntu 20.04 kernel 5.4.0-104 and containerd 1.4.12

### Installation method

Other (please describe above)

### Browser

_No response_

### Configuration changes

paperless-ngx is running in Kubernetes

### Other

_No response_

### PR DESCRIPTION
**Pull request by stumpylog**
## Proposed change

When starting the Docker container, adds a wait for the Redis broker to be up and ready.  Generally this sort of thing is taken care of by the `depends_on`, but certain ways of running don't have that functionality.

Why Python?  Mostly because we already include the redis library, so it doesn't require installing a new package just for this startup functionality.

Fixes #780

Passing logs
```
paperless-ngx-dev-webserver  | Waiting for Redis: redis://paperless-ngx-dev-broker:6379
paperless-ngx-dev-webserver  | Connected to Redis broker: redis://paperless-ngx-dev-broker:6379
```

Failing logs
```
paperless-ngx-dev-webserver  | Waiting for Redis: redis://paperless-ngx-dev-broker:6378
paperless-ngx-dev-webserver  | Redis ping #0 failed, waiting 5s
paperless-ngx-dev-webserver  | Redis ping #1 failed, waiting 5s
paperless-ngx-dev-webserver  | Redis ping #2 failed, waiting 5s
paperless-ngx-dev-webserver  | Redis ping #3 failed, waiting 5s
paperless-ngx-dev-webserver  | Redis ping #4 failed, waiting 5s
paperless-ngx-dev-webserver  | Failed to connect to: redis://paperless-ngx-dev-broker:6378
paperless-ngx-dev-webserver exited with code 1
```

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docker/wait-for-redis.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    attempt = 0
+    with Redis.from_url(url=REDIS_URL) as client:
+        while attempt < MAX_RETRY_COUNT:
+            try:
+                client.ping()
+                break
+            except Exception:
+                print(
+                    f"Redis ping #{attempt} failed, waiting {RETRY_SLEEP_SECONDS}s",
+                    flush=True,
```
Comment (bauerj) [inline docker/wait-for-redis.py:32]: Why does this need `flush=True`?

File: docker/wait-for-redis.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                )
+                time.sleep(RETRY_SLEEP_SECONDS)
+                attempt += 1
+
+    if attempt >= MAX_RETRY_COUNT:
+        print(f"Failed to connect to: {REDIS_URL}")
```
Comment (bauerj) [inline docker/wait-for-redis.py:38]: And this not?

File: docker/wait-for-redis.py (hunk 1 above)
Comment (stumpylog) [reply docker/wait-for-redis.py:32]: Without that, the status messages didn't show up in a timely fashion, which might lead to confusion about what, if anything, the image is doing.  Some sort of buffering I assume

File: docker/wait-for-redis.py (hunk 2 above)
Comment (stumpylog) [reply docker/wait-for-redis.py:38]: At this point, the next this is exiting, which will flush stdout already, so a flush would be ok, but a little redundant.

**Review by bauerj — APPROVED** [review]

**Review by qcasey — APPROVED** [review]

**Pull request merged**